### PR TITLE
build: Remove UPCOMING.md

### PR DESCRIPTION
**Pull request by tylerbutler**
Removes the UPCOMING.md file and associated build tasks.

As implemented, the file can conflict a lot even with main branch commits (since there are changesets added all the time), and there is conflicting information about whether partners actually find a roll up of upcoming _unreleased_ changes useful.

The file can still be generated manually if needed. This might be useful for the release driver to create release notes. Also, we will still create changelogs for each package. This change only affects upcoming.md.

We can also still consider producing a page for the website with this info.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,7 +47,6 @@
 		"format:changed:main": "pretty-quick --branch main",
 		"format:changed:next": "pretty-quick --branch next",
 		"format:repo": "npm run prettier:repo:fix",
-		"generate:upcoming": "flub generate upcoming --releaseGroup client -t minor",
```
Comment (anthony-murphy) [inline package.json:50]: wouldn't we still want this so release oce can generate the roll up at release time?

File: build-tools/packages/build-cli/src/commands/generate/changeset.ts [hunk 2]
```diff
@@ -285,9 +285,6 @@ export default class GenerateChangesetCommand extends BaseCommand<typeof Generat
 		);
 		const changesetPath = path.relative(context.gitRepo.resolvedRoot, newFile);
 
-		// Update the UPCOMING.md file
-		await GenerateUpcomingCommand.run(["--releaseGroup", "client", "--releaseType", bumpType]);
```
Comment (anthony-murphy) [inline build-tools/packages/build-cli/src/commands/generate/changeset.ts:289]: wouldn't we still want this so release oce can generate the roll up at release time?

File: package.json (hunk 1 above)
Comment (tylerbutler) [reply package.json:50]: We don't usually have scripts for release-related activities. The release drivers run `flub` directly or via `pnpm flub`, so no need to define scripts for things the release drivers do.

**Review by anthony-murphy — APPROVED** [review]

File: build-tools/packages/build-cli/src/commands/generate/changeset.ts (hunk 2 above)
Comment (tylerbutler) [reply build-tools/packages/build-cli/src/commands/generate/changeset.ts:290]: This change removes the automatic/implicit generation of UPCOMING.md when a changeset is created using the CLI. If left in place, UPCOMING.md would continue to automatically generate each time a new changeset is created.

**Pull request merged**